core/eolearn/core/eodata.py

```diff
@@ -1030,6 +1080,38 @@ def _parse_feature_value(self, value):
         return value
 
 
+class _Loader:
+    def __init__(self, filesystem, ftype, fname, path):
+        self.filesystem = filesystem
+        self.ftype = ftype
+        self.fname = fname
+        self.path = path
+
+    def load(self):
+        file_handle = BytesIO()
+        self.filesystem.download(self.path, file_handle)
+        file_handle.seek(0)
+
+        if self.path.endswith('.gz'):
+            with gzip.open(file_handle, 'rb') as gzip_fp:
+                return self._decode(gzip_fp, self.path)
+
+        return self._decode(file_handle, self.path)
```

Comment:
As discussed a few moments ago, we could get rid of intermediate `BytesIO` and do the following:
```Python
        with self.filesystem.open(self.path, 'rb') as file_handle:
            if self.path.endswith('.gz'):
                with gzip.open(file_handle, 'rb') as gzip_fp:
                    return self._decode(gzip_fp, self.path)

            return self._decode(file_handle, self.path)
```
This way we might avoid some memory duplications.

Reply:
Actually, instead of `self.filesystem.open(self.path, 'rb')` we could even call `self.filesystem.openbin(self.path)` as `open` is only a wrapper around `openbin`.

Reply:
Done.